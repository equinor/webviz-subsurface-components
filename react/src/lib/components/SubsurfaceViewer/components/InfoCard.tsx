import React from "react";
import {
    Collapse,
    Table,
    TableBody,
    TableCell,
    TableContainer,
    TableRow,
} from "@mui/material";
import { Button, Icon } from "@equinor/eds-core-react";
import { arrow_drop_up, arrow_drop_down } from "@equinor/eds-icons";
import { styled } from "@mui/system";

import {
    ExtendedLayerProps,
    LayerPickInfo,
    PropertyDataType,
} from "../layers/utils/layerTools";
import { rgb } from "d3-color";
import { FeatureCollection } from "geojson";

Icon.add({ arrow_drop_up, arrow_drop_down });

interface InfoCardDataType {
    layerName: string;
    properties?: PropertyDataType[];
}

export interface InfoCardProps {
    /**
     * List of JSON object describing picking information of layers
     * that are under the cursor.
     */
    pickInfos: LayerPickInfo[];
}

const roundToSignificant = function (num: number) {
    // Returns two significant figures (non-zero) for numbers with an absolute value less
    // than 1, and two decimal places for numbers with an absolute value greater
    // than 1.
    return parseFloat(
        num.toExponential(Math.max(1, 2 + Math.log10(Math.abs(num))))
    );
};

const StyledTable = styled(Table)({
    "& > *": {
        backgroundColor: "#ffffffcc",
        color: "#000000ff",
        border: "2px solid #ccc",
        padding: "0px",
        borderRadius: "5px",
        position: "absolute",
        bottom: 0,
        left: 0,
        marginLeft: "3px",
        marginBottom: "3px",
    },
});

const StyledTableCell = styled(TableCell)({
    border: "none",
    padding: 0,
    width: "20px",
});

const StyledTableRow = styled(TableRow)({
    "& > *": {
        padding: 0,
    },
});

function Row(props: { layer_data: InfoCardDataType }) {
    const { layer_data } = props;
    const [open, setOpen] = React.useState(true);

    if (layer_data.properties?.length == 0) return null;
    return (
        <React.Fragment>
            <StyledTableRow>
                <StyledTableCell>
                    <Button
                        style={{ padding: 0 }}
                        variant="ghost"
                        onClick={() => setOpen(!open)}
                    >
                        {open ? (
                            <Icon color="currentColor" name="arrow_drop_up" />
                        ) : (
                            <Icon color="currentColor" name="arrow_drop_down" />
                        )}
                    </Button>
                </StyledTableCell>
                <TableCell> {layer_data.layerName} </TableCell>
            </StyledTableRow>
            <StyledTableRow>
                <TableCell
                    style={{ paddingBottom: 0, paddingTop: 0 }}
                    colSpan={2}
                >
                    <Collapse in={open} timeout="auto" unmountOnExit>
                        <Table size="small" aria-label="properties">
                            <TableBody>
                                {layer_data.properties?.map((propertyRow) => (
                                    <StyledTableRow key={propertyRow.name}>
                                        <TableCell
                                            style={{
                                                border: "none",
                                                paddingLeft: 10,
                                                paddingRight: 10,
                                            }}
                                        >
                                            {propertyRow.color && (
                                                <span
                                                    style={{
                                                        color: rgb(
                                                            ...(propertyRow.color as [
                                                                number,
                                                                number,
                                                                number,
                                                                number?
                                                            ])
                                                        ).toString(),
                                                    }}
                                                >
                                                    {"\u2B24"}
                                                </span>
                                            )}
                                            {propertyRow.name}
                                        </TableCell>
                                        <TableCell
                                            style={{
                                                border: "none",
                                                textAlign: "right",
                                            }}
                                        >
                                            {typeof propertyRow.value ==
                                            "number"
                                                ? roundToSignificant(
                                                      propertyRow.value
                                                  )
                                                : propertyRow.value}
                                        </TableCell>
                                    </StyledTableRow>
                                ))}
                            </TableBody>
                        </Table>
                    </Collapse>
                </TableCell>
            </StyledTableRow>
        </React.Fragment>
    );
}

const InfoCard: React.FC<InfoCardProps> = (props: InfoCardProps) => {
    const [infoCardData, setInfoCardData] = React.useState<
        InfoCardDataType[] | null
    >(null);

    React.useEffect(() => {
        if (props.pickInfos.length === 0) {
            setInfoCardData(null);
            return;
        }
        const topObject = props.pickInfos[0];
        if (
            topObject.coordinate === undefined ||
            topObject.coordinate.length < 2
        ) {
            return;
        }

        const xy_properties: PropertyDataType[] = [];
        xy_properties.push({
            name: "x",
            value: Number(topObject.coordinate[0]).toFixed(2).toString() + " m",
        });
        xy_properties.push({
            name: "y",
            value: Number(topObject.coordinate[1]).toFixed(2).toString() + " m",
        });

        const info_card_data: InfoCardDataType[] = [];
        info_card_data.push({
            layerName: "Position",
            properties: xy_properties,
        });

        props.pickInfos.forEach((info) => {
            const layer_properties = info.properties;
            const layer_name = (
                info.layer
                    ?.props as unknown as ExtendedLayerProps<FeatureCollection>
            )?.name;

            // pick info can have 2 types of properties that can be displayed on the info card
            // 1. defined as propertyValue, used for general layer info (now using for positional data)
            // 2. Another defined as array of property object described by type PropertyDataType

            // collecting card data for 1st type
            const zValue = (info as LayerPickInfo).propertyValue;
            if (typeof zValue !== "undefined") {
                const property = xy_properties.find(
                    (item) => item.name === layer_name
                );
                if (property) {
                    property.value = zValue;
                } else {
                    xy_properties.push({
                        name: layer_name,
                        value: zValue,
                    });
                }
            }

            // collecting card data for 2nd type
            const layer = info_card_data.find(
                (item) => item.layerName === layer_name
            );
            if (layer) {
                layer_properties?.forEach((layer_prop) => {
                    const property = layer.properties?.find(
                        (item) => item.name === layer_prop.name
                    );
                    if (property) {
                        property.value = layer_prop.value;
                    } else {
                        layer.properties?.push(layer_prop);
                    }
                });
            } else {
                info_card_data.push({
                    layerName: layer_name || "unknown-layer",
                    properties: layer_properties,
                });
            }
        });

        setInfoCardData(info_card_data);
    }, [props.pickInfos]);

    return (
        infoCardData && (
            <TableContainer>
                <StyledTable aria-label="info-card">
                    <TableBody>
                        {infoCardData.map(
                            (card_data) =>
                                card_data.properties && (
                                    <Row
                                        key={card_data.layerName}
                                        layer_data={card_data}
                                    />
                                )
                        )}
                    </TableBody>
                </StyledTable>
            </TableContainer>
        )
    );
};

export default InfoCard;
