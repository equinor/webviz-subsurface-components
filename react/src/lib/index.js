/* This Source Code Form is subject to the terms of the Mozilla Public
 * License, v. 2.0. If a copy of the MPL was not distributed with this
 * file, You can obtain one at https://mozilla.org/MPL/2.0/.
 *
 * Copyright (C) 2020 - Equinor ASA. */

import DeckGLMap from "./components/DeckGLMap";
import GroupTree from "./components/GroupTree";
import HistoryMatch from "./components/HistoryMatch";
import LayeredMap from "./components/LayeredMap";
import LeafletMap from "./components/LeafletMap";
import Map from "./components/Map";
import Morris from "./components/Morris";
import PriorPosteriorDistribution from "./components/PriorPosteriorDistribution";
import VectorSelector from "./components/VectorSelector";
import WellCompletions from "./components/WellCompletions";
import WellLogViewer from "./components/WellLogViewer";

export {
    HistoryMatch,
    Morris,
    Map,
    LayeredMap,
    PriorPosteriorDistribution,
    LeafletMap,
    DeckGLMap,
    VectorSelector,
    WellCompletions,
    GroupTree,
    WellLogViewer,
};
