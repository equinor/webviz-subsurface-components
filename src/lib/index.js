import HistoryMatch from "./components/HistoryMatch";
import Map from "./components/Map";
import Morris from "./components/Morris";
import LayeredMap from "./components/LayeredMap";
import LeafletMap from "./components/LeafletMap";
import PriorPosteriorDistribution from "./components/PriorPosteriorDistribution";

export {
    HistoryMatch,
    Morris,
    Map,
    LayeredMap,
    PriorPosteriorDistribution,
    LeafletMap,
};
