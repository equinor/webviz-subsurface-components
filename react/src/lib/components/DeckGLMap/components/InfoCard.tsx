import React from "react";
import { makeStyles } from "@material-ui/core/styles";
import Collapse from "@material-ui/core/Collapse";
import Table from "@material-ui/core/Table";
import TableBody from "@material-ui/core/TableBody";
import TableCell from "@material-ui/core/TableCell";
import TableContainer from "@material-ui/core/TableContainer";
import TableRow from "@material-ui/core/TableRow";
import Paper from "@material-ui/core/Paper";

import { Button, Icon } from "@equinor/eds-core-react";
import { arrow_drop_up, arrow_drop_down } from "@equinor/eds-icons";

import { PickInfo } from "@deck.gl/core/lib/deck";
import { LayerPickInfo, PropertyDataType } from "../layers/utils/layerTools";
import { PropertyMapPickInfo } from "../layers/utils/propertyMapTools";
import { rgb } from "d3-color";

Icon.add({ arrow_drop_up, arrow_drop_down });

interface InfoCardDataType {
    layerName: string;
    properties: PropertyDataType[];
}

export interface InfoCardProps {
    /**
     * List of JSON object describing picking information of layers
     * that are under the cursor.
     */
    pickInfos: PickInfo<unknown>[];
}

const useStyles = makeStyles({
    table: {
        "& > *": {
            backgroundColor: "#ffffffcc",
            border: "2px solid #ccc",
            padding: "0px",
            borderRadius: "5px",

            position: "absolute",
            bottom: 0,
        },
    },
    icon_style: {
        border: "none",
        padding: 0,
        width: "20px",
    },
    table_row: {
        "& > *": {
            padding: 0,
        },
    },
});

function Row(props: { layer_data: InfoCardDataType }) {
    const { layer_data } = props;
    const [open, setOpen] = React.useState(true);
    const classes = useStyles();

    return (
        <React.Fragment>
            <TableRow className={classes.table_row}>
                <TableCell className={classes.icon_style}>
                    <Button
                        style={{ padding: 0 }}
                        variant="ghost"
                        onClick={() => setOpen(!open)}
                    >
                        {open ? (
                            <Icon color="currentColor" name="arrow_drop_up" />
                        ) : (
                            <Icon color="currentColor" name="arrow_drop_down" />
                        )}
                    </Button>
                </TableCell>
                <TableCell> {layer_data.layerName} </TableCell>
            </TableRow>
            <TableRow className={classes.table_row}>
                <TableCell
                    style={{ paddingBottom: 0, paddingTop: 0 }}
                    colSpan={2}
                >
                    <Collapse in={open} timeout="auto" unmountOnExit>
                        <Table size="small" aria-label="properties">
                            <TableBody>
                                {layer_data.properties?.map((propertyRow) => (
                                    <TableRow
                                        key={propertyRow.name}
                                        className={classes.table_row}
                                    >
                                        <TableCell
                                            style={{
                                                border: "none",
                                                paddingLeft: 10,
                                                paddingRight: 10,
                                            }}
                                        >
                                            {propertyRow.color && (
                                                <span
                                                    style={{
                                                        color: rgb(
                                                            ...propertyRow.color
                                                        ).toString(),
                                                    }}
                                                >
                                                    {"\u2B24"}
                                                </span>
                                            )}
                                            {propertyRow.name}
                                        </TableCell>
                                        <TableCell
                                            style={{
                                                border: "none",
                                                textAlign: "right",
                                            }}
                                        >
                                            {typeof propertyRow.value ==
                                            "number"
                                                ? propertyRow.value?.toFixed(2)
                                                : propertyRow.value}
                                        </TableCell>
                                    </TableRow>
                                ))}
                            </TableBody>
                        </Table>
                    </Collapse>
                </TableCell>
            </TableRow>
        </React.Fragment>
    );
}

const InfoCard: React.FC<InfoCardProps> = (props: InfoCardProps) => {
    const [infoCardData, setInfoCardData] =
        React.useState<InfoCardDataType[] | null>(null);

    React.useEffect(() => {
        if (props.pickInfos.length === 0) {
            setInfoCardData(null);
            return;
        }

        const topObject = props.pickInfos[0];
        if (
            topObject.coordinate === undefined ||
            topObject.coordinate.length < 2
        ) {
            return;
        }

        const xy_properties: PropertyDataType[] = [];
        xy_properties.push({ name: "x", value: topObject.coordinate[0] });
        xy_properties.push({ name: "y", value: topObject.coordinate[1] });

        const infoCardData: InfoCardDataType[] = [];
        infoCardData.push({
            layerName: "Position",
            properties: xy_properties,
        });

        props.pickInfos.forEach((info) => {
            const layer_properties = (info as LayerPickInfo)?.properties;
            const parent = infoCardData.find(
                (item) => item.layerName === info.layer?.id
            );
            if (parent) {
                layer_properties?.forEach((layer_prop) => {
                    const property = parent.properties.find(
                        (item) => item.name === layer_prop.name
                    );
                    if (property) {
                        property.value = layer_prop.value;
                    } else {
                        parent.properties.push(layer_prop);
                    }
                });
            } else {
                infoCardData.push({
                    layerName: info.layer?.id || "unknown-layer",
                    properties: layer_properties,
                });
            }

            const zValue = (info as PropertyMapPickInfo).propertyValue;
            if (zValue) {
                const property = xy_properties.find(
                    (item) => item.name === info.layer.id
                );
                if (property) {
                    property.value = zValue;
                } else {
                    xy_properties.push({ name: info.layer.id, value: zValue });
                }
            }
        });

        setInfoCardData(infoCardData);
    }, [props.pickInfos]);

    const classes = useStyles();
    return (
        infoCardData && (
            <TableContainer component={Paper}>
                <Table aria-label="info-card" className={classes.table}>
                    <TableBody>
                        {infoCardData.map(
                            (card_data) =>
                                card_data.properties && (
                                    <Row
                                        key={card_data.layerName}
                                        layer_data={card_data}
                                    />
                                )
                        )}
                    </TableBody>
                </Table>
            </TableContainer>
        )
    );
};

export default InfoCard;
