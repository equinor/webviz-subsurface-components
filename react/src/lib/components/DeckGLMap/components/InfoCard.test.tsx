import {
    render,
    screen,
    waitForElementToBeRemoved,
} from "@testing-library/react";
import "jest-styled-components";
import "@testing-library/jest-dom";
import React from "react";
import InfoCard from "./InfoCard";
import userEvent from "@testing-library/user-event";
import { Layer, PickInfo } from "deck.gl";
import { LayerProps } from "@deck.gl/core/lib/layer";

describe("Test Info Card", () => {
    it("snapshot test with no props", () => {
        const { container } = render(
            <InfoCard
                pickInfos={[
                    {
                        x: 152,
                        y: 254,
                        radius: 1,
                        depth: 638,
                        coordinate: [111, 222],
                    } as unknown as PickInfo<unknown>,
                    {
                        layer: { id: "wells-layer" } as Layer<
                            unknown,
                            LayerProps<unknown>
                        >,
                        property: {
                            name: "Poro WellA",
                            value: 123,
                        },
                    } as unknown as PickInfo<unknown>,
                ]}
            />
        );
        expect(container.firstChild).toMatchSnapshot();
    });
    it("collapse infocard", async () => {
        render(
            <InfoCard
                pickInfos={[
                    {
                        x: 152,
                        y: 254,
                        radius: 1,
                        depth: 638,
                        coordinate: [111, 222],
                    } as unknown as PickInfo<unknown>,
                ]}
            />
        );
        const collapse_button = screen.getByRole("button", { name: "" });
        expect(screen.getByText("111")).toBeVisible();
        userEvent.click(collapse_button);
        await waitForElementToBeRemoved(() => screen.getByText("111"));
    });
    it("undefined coordinates", async () => {
        const { container } = render(
            <InfoCard
                pickInfos={[
                    {
                        x: 152,
                        y: 254,
                        // @ts-expect-error: to be fixed
                        radius: 1,
                        depth: 638,
                        coordinate: undefined,
                    },
                ]}
            />
        );
        expect(container.firstChild).toMatchSnapshot();
    });
});
