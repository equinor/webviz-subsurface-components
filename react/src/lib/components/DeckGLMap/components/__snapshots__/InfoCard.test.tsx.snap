// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Info Card snapshot test when property value provided 1`] = `
.c1 {
  display: grid;
  grid-gap: 8px;
  grid-auto-flow: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 100%;
}

.c0 {
  margin: 0;
  padding: 0;
  background: transparent;
  height: 36px;
  padding-left: 16px;
  padding-right: 16px;
  border: 1px solid transparent;
  border-radius: 4px;
  margin: 0;
  color: var(--eds_interactive_primary__resting,rgba(0,112,121,1));
  font-family: Equinor;
  font-size: 0.875rem;
  font-weight: 500;
  line-height: 1.143em;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  position: relative;
  cursor: pointer;
  display: inline-block;
}

.c0 svg {
  justify-self: center;
}

.c0::after {
  position: absolute;
  top: -6px;
  left: -0;
  width: 100%;
  height: 48px;
  content: '';
}

.c0:hover {
  background: var(--eds_interactive_primary__hover_alt,rgba(222,237,238,1));
  color: var(--eds_interactive_primary__hover,rgba(0,79,85,1));
  border: 1px solid transparent;
  border-radius: 4px;
}

.c0:focus {
  outline: none;
}

.c0[data-focus-visible-added]:focus {
  outline: 1px dashed rgba(0,112,121,1);
  outline-offset: 2px;
}

.c0::-moz-focus-inner {
  border: 0;
}

.c0:disabled {
  cursor: not-allowed;
  background: transparent;
  border: 1px solid transparent;
  margin: 0;
  color: var(--eds_interactive__disabled__text,rgba(190,190,190,1));
  font-family: Equinor;
  font-size: 0.875rem;
  font-weight: 500;
  line-height: 1.143em;
  text-align: center;
}

.c0:disabled:hover {
  background: transparent;
}

.c0::before {
  position: absolute;
  top: 0;
  left: 0;
  width: auto;
  min-height: auto;
  content: '';
}

<div
  class="MuiTableContainer-root"
>
  <table
    aria-label="info-card"
    class="MuiTable-root makeStyles-table-10"
  >
    <tbody
      class="MuiTableBody-root"
    >
      <tr
        class="MuiTableRow-root makeStyles-table_row-12"
      >
        <td
          class="MuiTableCell-root MuiTableCell-body makeStyles-icon_style-11"
        >
          <button
            class="c0"
            style="padding: 0px;"
            tabindex="0"
            type="button"
          >
            <span
              class="c1"
            >
              <svg
                aria-hidden="true"
                class=""
                fill="currentColor"
                height="24px"
                viewBox="0 0 24 24"
                width="24px"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  class=""
                  clip-rule="evenodd"
                  d="m7 14.5 5-5 5 5H7z"
                  data-testid="eds-icon-path"
                  fill-rule="evenodd"
                  height="24"
                />
              </svg>
            </span>
          </button>
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-body"
        >
           
          Position
           
        </td>
      </tr>
      <tr
        class="MuiTableRow-root makeStyles-table_row-12"
      >
        <td
          class="MuiTableCell-root MuiTableCell-body"
          colspan="2"
          style="padding-bottom: 0px; padding-top: 0px;"
        >
          <div
            class="MuiCollapse-root MuiCollapse-entered"
            style="min-height: 0px;"
          >
            <div
              class="MuiCollapse-wrapper"
            >
              <div
                class="MuiCollapse-wrapperInner"
              >
                <table
                  aria-label="properties"
                  class="MuiTable-root"
                >
                  <tbody
                    class="MuiTableBody-root"
                  >
                    <tr
                      class="MuiTableRow-root makeStyles-table_row-12"
                    >
                      <td
                        class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
                        style="padding-left: 10px; padding-right: 10px;"
                      >
                        x
                      </td>
                      <td
                        class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
                        style="text-align: right;"
                      >
                        111.00 m
                      </td>
                    </tr>
                    <tr
                      class="MuiTableRow-root makeStyles-table_row-12"
                    >
                      <td
                        class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
                        style="padding-left: 10px; padding-right: 10px;"
                      >
                        y
                      </td>
                      <td
                        class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
                        style="text-align: right;"
                      >
                        222.00 m
                      </td>
                    </tr>
                    <tr
                      class="MuiTableRow-root makeStyles-table_row-12"
                    >
                      <td
                        class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
                        style="padding-left: 10px; padding-right: 10px;"
                      >
                        Hill shading
                      </td>
                      <td
                        class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
                        style="text-align: right;"
                      >
                        3152.02
                      </td>
                    </tr>
                  </tbody>
                </table>
              </div>
            </div>
          </div>
        </td>
      </tr>
      <tr
        class="MuiTableRow-root makeStyles-table_row-12"
      >
        <td
          class="MuiTableCell-root MuiTableCell-body makeStyles-icon_style-11"
        >
          <button
            class="c0"
            style="padding: 0px;"
            tabindex="0"
            type="button"
          >
            <span
              class="c1"
            >
              <svg
                aria-hidden="true"
                class=""
                fill="currentColor"
                height="24px"
                viewBox="0 0 24 24"
                width="24px"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  class=""
                  clip-rule="evenodd"
                  d="m7 14.5 5-5 5 5H7z"
                  data-testid="eds-icon-path"
                  fill-rule="evenodd"
                  height="24"
                />
              </svg>
            </span>
          </button>
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-body"
        >
           
          Wells layer
           
        </td>
      </tr>
      <tr
        class="MuiTableRow-root makeStyles-table_row-12"
      >
        <td
          class="MuiTableCell-root MuiTableCell-body"
          colspan="2"
          style="padding-bottom: 0px; padding-top: 0px;"
        >
          <div
            class="MuiCollapse-root MuiCollapse-entered"
            style="min-height: 0px;"
          >
            <div
              class="MuiCollapse-wrapper"
            >
              <div
                class="MuiCollapse-wrapperInner"
              >
                <table
                  aria-label="properties"
                  class="MuiTable-root"
                >
                  <tbody
                    class="MuiTableBody-root"
                  >
                    <tr
                      class="MuiTableRow-root makeStyles-table_row-12"
                    >
                      <td
                        class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
                        style="padding-left: 10px; padding-right: 10px;"
                      >
                        Poro
                      </td>
                      <td
                        class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
                        style="text-align: right;"
                      >
                        123
                      </td>
                    </tr>
                    <tr
                      class="MuiTableRow-root makeStyles-table_row-12"
                    >
                      <td
                        class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
                        style="padding-left: 10px; padding-right: 10px;"
                      >
                        Perm
                      </td>
                      <td
                        class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
                        style="text-align: right;"
                      >
                        456
                      </td>
                    </tr>
                  </tbody>
                </table>
              </div>
            </div>
          </div>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`Test Info Card snapshot test with no props 1`] = `
.c1 {
  display: grid;
  grid-gap: 8px;
  grid-auto-flow: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 100%;
}

.c0 {
  margin: 0;
  padding: 0;
  background: transparent;
  height: 36px;
  padding-left: 16px;
  padding-right: 16px;
  border: 1px solid transparent;
  border-radius: 4px;
  margin: 0;
  color: var(--eds_interactive_primary__resting,rgba(0,112,121,1));
  font-family: Equinor;
  font-size: 0.875rem;
  font-weight: 500;
  line-height: 1.143em;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  position: relative;
  cursor: pointer;
  display: inline-block;
}

.c0 svg {
  justify-self: center;
}

.c0::after {
  position: absolute;
  top: -6px;
  left: -0;
  width: 100%;
  height: 48px;
  content: '';
}

.c0:hover {
  background: var(--eds_interactive_primary__hover_alt,rgba(222,237,238,1));
  color: var(--eds_interactive_primary__hover,rgba(0,79,85,1));
  border: 1px solid transparent;
  border-radius: 4px;
}

.c0:focus {
  outline: none;
}

.c0[data-focus-visible-added]:focus {
  outline: 1px dashed rgba(0,112,121,1);
  outline-offset: 2px;
}

.c0::-moz-focus-inner {
  border: 0;
}

.c0:disabled {
  cursor: not-allowed;
  background: transparent;
  border: 1px solid transparent;
  margin: 0;
  color: var(--eds_interactive__disabled__text,rgba(190,190,190,1));
  font-family: Equinor;
  font-size: 0.875rem;
  font-weight: 500;
  line-height: 1.143em;
  text-align: center;
}

.c0:disabled:hover {
  background: transparent;
}

.c0::before {
  position: absolute;
  top: 0;
  left: 0;
  width: auto;
  min-height: auto;
  content: '';
}

<div
  class="MuiTableContainer-root"
>
  <table
    aria-label="info-card"
    class="MuiTable-root makeStyles-table-1"
  >
    <tbody
      class="MuiTableBody-root"
    >
      <tr
        class="MuiTableRow-root makeStyles-table_row-3"
      >
        <td
          class="MuiTableCell-root MuiTableCell-body makeStyles-icon_style-2"
        >
          <button
            class="c0"
            style="padding: 0px;"
            tabindex="0"
            type="button"
          >
            <span
              class="c1"
            >
              <svg
                aria-hidden="true"
                class=""
                fill="currentColor"
                height="24px"
                viewBox="0 0 24 24"
                width="24px"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  class=""
                  clip-rule="evenodd"
                  d="m7 14.5 5-5 5 5H7z"
                  data-testid="eds-icon-path"
                  fill-rule="evenodd"
                  height="24"
                />
              </svg>
            </span>
          </button>
        </td>
        <td
          class="MuiTableCell-root MuiTableCell-body"
        >
           
          Position
           
        </td>
      </tr>
      <tr
        class="MuiTableRow-root makeStyles-table_row-3"
      >
        <td
          class="MuiTableCell-root MuiTableCell-body"
          colspan="2"
          style="padding-bottom: 0px; padding-top: 0px;"
        >
          <div
            class="MuiCollapse-root MuiCollapse-entered"
            style="min-height: 0px;"
          >
            <div
              class="MuiCollapse-wrapper"
            >
              <div
                class="MuiCollapse-wrapperInner"
              >
                <table
                  aria-label="properties"
                  class="MuiTable-root"
                >
                  <tbody
                    class="MuiTableBody-root"
                  >
                    <tr
                      class="MuiTableRow-root makeStyles-table_row-3"
                    >
                      <td
                        class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
                        style="padding-left: 10px; padding-right: 10px;"
                      >
                        x
                      </td>
                      <td
                        class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
                        style="text-align: right;"
                      >
                        111.00 m
                      </td>
                    </tr>
                    <tr
                      class="MuiTableRow-root makeStyles-table_row-3"
                    >
                      <td
                        class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
                        style="padding-left: 10px; padding-right: 10px;"
                      >
                        y
                      </td>
                      <td
                        class="MuiTableCell-root MuiTableCell-body MuiTableCell-sizeSmall"
                        style="text-align: right;"
                      >
                        222.00 m
                      </td>
                    </tr>
                  </tbody>
                </table>
              </div>
            </div>
          </div>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;

exports[`Test Info Card undefined coordinates 1`] = `null`;
